### ISSUE TITLE
[aws-efs] allow mounting by DNS and not FSID

### ISSUE THREAD

**Issue by djdevin** [kubernetes-incubator/external-storage#422]
We have a CNAME mapped to our EFS DNS name for ease of use and cross-environment usage (so nfs.ENVIRONMENT_HOST.com will always point to the correct place)

This might be a Linux or Openshift issue because it does something weird where even though we didn't pass the DNS name, the DNS name was mounted into this pod. I think it's going to use it because it was previously mounted somewhere on that host machine. Running the provisioner on a machine with no NFS mounts is fine. Take a look at this

```
F1025 22:26:23.928789       1 efs-provisioner.go:71] no mount entry found for fs-ad138fe4.efs.us-east-1.amazonaws.com among entries overlay:/, proc:/proc, tmpfs:/dev,
[snip] 
mqueue:/dev/mqueue, nfs.test.env.example.com:/persistentvolumes:/persistentvolumes, /dev/mapper/atomicos-root:/dev/termination-log,
```

So as we can see it's definitely mounted but under a different DNS name, it will never find it.

**Comment by wongma7** [comment]
Yes, I will add a way for you to specify the DNS name. I guess via ENV var. 

Idea behind this strict checking of the mount entry is paranoia that we don't touch something local d or some other nfs mount if the mount point was e.g. mistyped

**Comment by OverlordQ** [comment]
Need this to work with specifying IPs also.  We're not leveraging the Amazon provided DNS.

**Comment by wongma7** [comment]
If you don't need these safeguards there is no practical difference between the efs provisioner and nfs-client https://github.com/kubernetes-incubator/external-storage/tree/master/nfs-client . You can specify mountOptions for nfs-client now and there isn't currently a way for efs provisioner to leverage the efs mount helper (see https://github.com/kubernetes-incubator/external-storage/issues/725) so nfs-client can be a drop-in replacement 100% of the time https://github.com/kubernetes-incubator/external-storage/tree/master/nfs-client

**Comment by ReillyProcentive** [comment]
Would the out-of-tree CSI volume plugin feature help with this? [link](https://github.com/kubernetes/features/issues/178)
